openassessment/xblock/openassessmentblock.py

```diff
@@ -364,6 +364,82 @@ def student_view(self, context=None):
 
+        return self._update_and_return_fragment(fragment, 'OpenAssessmentBlock')
+
+    def ora_blocks_listing_view(self, context=None):
+        """This view is used in the Open Response Assessment tab in the LMS Instructor Dashboard
+        to display all available course ORA blocks.
+
+        Args:
+            context: contains two items:
+                "ora_items" - all course items with names and parents, example:
+                    [{"parent_name": "Vertical name",
+                      "name": "ORA Display Name",
+                      "url_grade_available_responses": "/grade_available_responses_view",
+                      "staff_assessment": false,
+                      "parent_id": "vertical_block_id",
+                      "url_base": "/student_view",
+                      "id": "openassessment_block_id"
+                     }, ...]
+                "ora_item_view_enabled" - enabled LMS API endpoint to serve XBlock view or not
+
+        Returns:
+            (Fragment): The HTML Fragment for this XBlock.
+        """
+        ora_items = context.get('ora_items', []) if context else []
+        ora_item_view_enabled = context.get('ora_item_view_enabled', False) if context else False
+        context_dict = {
+            "ora_items": json.dumps(ora_items),
+            "ora_item_view_enabled": ora_item_view_enabled
+        }
+
+        template = get_template('openassessmentblock/instructor_dashboard/oa_listing.html')
+        context = Context(context_dict)
+        fragment = Fragment(template.render(context))
+
+        return self._update_and_return_fragment(fragment, 'CourseOpenResponsesListingBlock')
+
+    def grade_available_responses_view(self, context=None):
+        """Grade Available Responses view.
+
+        Auxiliary view which displays the staff grading area
+        (used in the Open Response Assessment tab in the Instructor Dashboard of LMS)
+
+        Args:
+            context: Not used for this view.
+
+        Returns:
+            (Fragment): The HTML Fragment for this XBlock.
+        """
+        student_item = self.get_student_item_dict()
+        staff_assessment_required = "staff-assessment" in self.assessment_steps
+
+        context_dict = {
+            "title": self.title,
+            'staff_assessment_required': staff_assessment_required
+        }
+
+        if staff_assessment_required:
+            context_dict.update(
+                self.get_staff_assessment_statistics_context(student_item["course_id"], student_item["item_id"])
+            )
+
+        template = get_template('openassessmentblock/instructor_dashboard/oa_grade_available_responses.html')
+        context = Context(context_dict)
+        fragment = Fragment(template.render(context))
+
+        return self._update_and_return_fragment(fragment, 'StaffAssessmentBlock')
+
+    def _update_and_return_fragment(self, fragment, initialize_js_func, additional_css=None, additional_js=None):
```

Comment:
I find this function confusing. I recommend that instead of taking the fragment, this function should take the HTML, since in all three cases the fragment is constructed from the HTML. You could then name the method ``create_fragment`` (or something like it).